File: 按钮组底层实现/buttonGroup.nim
var status = [0,0,0,0]
proc change(idx:int):array[0..3,int] = 
    status[idx] = 1
    for index, item in status:
        if index != idx:
            status[index] = 0
    return status
# 将第三个元素开启，同时其他元素关闭
echo(change(3))

File: 游戏引擎类算法/按钮组底层实现/buttonGroup.java
import java.util.Arrays;

public class buttonGroup
{
    private static int[] status = {1,1,1,1};
    public static int[] change(int idx)
    {
        status[idx] = 1;
        for(int i = 0; i < status.length; i++)
        {
            if(i != idx)
            {
                status[i] = 0;
            }
        }
        return status;
    }
    public static void main(String args[])
    {
        System.out.println(Arrays.toString(change(3)));
    }
}